pwndbg/commands/xinfo.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import gdb
+import subprocess
+
+import pwndbg.arch
+import pwndbg.commands
+import pwndbg.config
+import pwndbg.memory
+import pwndbg.vmmap
+import pwndbg.regs
+import pwndbg.color.memory as M
+
+
+parser = argparse.ArgumentParser(description='Shows offsets of the specified address to useful other locations')
+parser.add_argument('address', nargs='?', default='$pc', 
+                    help='Address to inspect')
+
+def print_line(name, addr, first, second, op, width = 20):
+    FMT = "{} {} = {} {} {:#x}"
+
+    print(FMT.format(
+        name.rjust(width),
+        M.get(addr),
+        M.get(first) if type(first) is not str else first.ljust(len(hex(addr))),
+        op,
+        second,))
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+def xinfo(address=None):
+    addr = int(address)
+    addr &= pwndbg.arch.ptrmask
+
+    page = pwndbg.vmmap.find(addr)
+
+    print("Extended information for virtual address {}:".format(M.get(addr)))
+
+    if page is None:
+        print("\n  Virtual address {:#x} is not mapped.".format(addr))
+        return
+
+    print("\n  Containing mapping:")
+    print(M.get(address, text=str(page)))
+
+    print("\n  Offset information:")
+
+    if page.objfile == "[stack]":
```

Comment:
Can we add a property to `Page` object called `is_stack` that would do exactly this?